lib/tasks/foreman_templates_tasks.rake

```diff
@@ -0,0 +1,82 @@
+# Tasks
+namespace :templates do
+  desc 'Synchronize templates from a git repo'
+  task :sync => :environment do
+    # Available options:
+    #* verbose   => Print extra information during the run [false]
+    #* repo      => Sync templates from a different Git repo [https://github.com/theforeman/community-templates]
+    #* branch    => Branch in Git repo [default branch]
+    #* prefix    => The string all imported templates should begin with [Community ]
+    #* dirname   => The directory within the git tree containing the templates [/]
+    #* filter    => Import names matching this regex (case-insensitive; snippets are not filtered)
+    #* associate => Associate to OS, always, new or never  [new]
+
+    ForemanTemplates::TemplateImporter.new({
+      verbose:   ENV['verbose'],
+      repo:      ENV['repo'],
+      branch:    ENV['branch'],
+      prefix:    ENV['prefix'],
+      dirname:   ENV['dirname'],
+      filter:    ENV['filter'],
+      associate: ENV['associate'],
+    }).import!
+  end
+
+  task :purge => :environment do
+    # Available options:
+    #* verbose => Print extra information during the run [false]
+    #* not     => negate prefix / filter
+    #* kind    => Template kind
+    #* prefix  => The string all imported templates should begin with [Community ]
+    #* filter  => Import names matching this regex (case-insensitive; snippets are not filtered)
+
+    @verbose = ( ENV['verbose'] and ENV['verbose'] != 'false' ) ? true : false
+    match    = ENV['not'] ? 'NOT LIKE' : 'LIKE'
+    kind     = ENV['kind'] ? ENV['kind'] : "template_kind #{match} '%'"
+    prefix   = ENV['prefix'] ? ENV['prefix'] : 'Community '
+    filter   = ENV['filter'] ? ENV['filter'] : "name #{match} '#{prefix}%'"
+
+    #clause = [filter, kind].join(" AND ")
+    ProvisioningTemplate.where(filter).each do |template|
+      puts template if @verbose
+      template.destroy
```

Comment:
This obviously works, but the point of #27 is to move the executable actions out of the Rakefile and into a service class. This is better because when I (eventually) write a UI for the plugin, I don't want to be calling Rake tasks from the UI - rather, both the UI and the Rake task should use a shared codebase

Could you move this to a trivial method in the Importer class (`app/services/foreman_templates/template_importer.rb`) so that it's available there? It should be a case of copying whats on line 14-22 here and calling `.purge!` or whatever you call the method